runtime/sema/type.go

```diff
@@ -3146,7 +3184,7 @@ func (t *FunctionType) Resolve(typeArguments *TypeParameterTypeOrderedMap) Type
 
 	// parameters
 
-	var newParameters []Parameter
+	var newParameters []Parameter = make([]Parameter, 0, len(t.Parameters))
```

Comment:
We shouldn't unnecessarily allocate an empty slice here if there are no parameters (`len(t.Parameters)`).

Maybe change the code here and elsewhere in the PR where `make` is called to the pattern we use elsewhere:
- Define the slice variable (causes no allocation)
- Check the length of the input. If positive:
  -  Pre-allocate
  - Perform operation

For example: https://github.com/onflow/cadence/blob/2081a601106baaf6ae695e3f2a84613160bb2166/runtime/convertTypes.go#L411-L420